# Exploratory Data Analysis of Address

In [ ]:
import pandas as pd

In [8]:
# Create a new DataFrame for films
address_df= pd.read_csv("../data/raw/unclean-address.csv")

# Display info about the dataframe
address_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 603 entries, 0 to 602
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   address_id   603 non-null    int64  
 1   address      603 non-null    object 
 2   address2     0 non-null      float64
 3   district     600 non-null    object 
 4   city_id      603 non-null    int64  
 5   postal_code  599 non-null    float64
 6   phone        601 non-null    float64
 7   last_update  603 non-null    object 
dtypes: float64(3), int64(2), object(3)
memory usage: 37.8+ KB


There are some missing values in `district`, `postal_code` and `phone` columns. 

In [ ]:
# Check for null values
address_df.isnull().any(axis=1).sum()

There are 603 rows with null values. This is because of the `address2` column which has no values. This can be safely dropped from the table as no data would be lost.

In [ ]:
# Check for any duplicates in the DataFrame
duplicates = address_df.duplicated().sum()
print(f"There are {duplicates} duplicate rows in the address DataFrame")

There are no duplicate roles in the address DataFrame so no need for cleaning.

In [7]:
# Show the different date formats 
unique_date_formats = address_df['last_update'].unique()
print(f'Different date formats in last_update: {unique_date_formats}') 

Different date formats in last_update: ['2021-04-08 14:36:58.200310']


In [9]:
# Show the different data types in the release_year column
release_year_types = address_df['city_id'].map(type).unique()
print(f'Different data types in city_id column: {release_year_types}')

Different data types in city_id column: [<class 'int'>]
